commit ea8a63a5668f523f5c65bbe466eaf80c4101f085
Author: John Levon <john.levon@joyent.com>
Date:   2019-03-11T19:49:01+00:00 (7 months ago)
    
    OS-7604 -faggressive-loop-optimizations should be off everywhere
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/Makefile b/Makefile
index 26bf5a3..ee2192b 100644
--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,8 @@
 # Copyright (c) 2019, Joyent, Inc.
 #
 
+include		$(PWD)/../../../build.env
+
 KERNEL_SOURCE =	$(PWD)/../../illumos
 MDB_SOURCE =	$(KERNEL_SOURCE)/usr/src/cmd/mdb
 PROTO_AREA =	$(PWD)/../../../proto
@@ -92,6 +94,13 @@ ALWAYS_CFLAGS = \
 	-Werror \
 	-fno-inline-functions
 
+#
+# Skip dangerous GCC options (not that any specific problems are know of here).
+#
+ifneq ($(PRIMARY_COMPILER_VER),4)
+ALWAYS_CFLAGS += -fno-aggressive-loop-optimizations
+endif
+
 #
 # Replacing -O with -O2 causes the KVM host to panic.  Don't do that.
 #
